### ISSUE TITLE
Support for snapshots

### ISSUE THREAD

**Issue by rajdip-b** [keyshade-xyz/keyshade#121]
## Description

We would like the users to enable the feature of snapshot on every workspace they **own or are administrators** in. A snapshot will create a restore point of the entire workspace at any point in time. This includes:
- roles
- environments
- secrets
- projects
- workspace details
- events

#### Snapshots can be created in the following ways:
- manually
- creating a cron job that will do it for them

#### Related functionalities:
- Delete snapshot
- Get a list of all snapshots
- Restore to a particular snapshot (doing so will remove all the snapshots mad after this point)

## Solution

- [ ] We would need to store the snapshots in JSON/YAML format in an S3 compatible storage device.
- [ ] Create a model in `schema.prisma` to store the details like name, createdOn, s3FileKey in our database.
- [ ] Create service and controller for implementing the business logic

## Additional context
Currently blocked by #120

**Comment by rajdip-b** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Snapshots](https://trello.com/c/9aVNuqpn/31-snapshots)

**Comment by luxmikant** [comment]
hello maintainer i would  like to work on this issue

**Comment by rajdip-b** [comment]
Hey @luxmikant, this issue is not a priority at the moment. So we don't really want our contributors to be spending time in this issue!
